pandas/core/frame.py

```diff
@@ -3385,15 +3385,9 @@ def sort_index(self, axis=0, level=None, ascending=True, inplace=False,
                                                  sort_remaining=sort_remaining)
 
         elif isinstance(labels, MultiIndex):
-            from pandas.core.sorting import lexsort_indexer
-
-            # make sure that the axis is lexsorted to start
-            # if not we need to reconstruct to get the correct indexer
-            if not labels.is_lexsorted():
-                labels = MultiIndex.from_tuples(labels.values)
 
-            indexer = lexsort_indexer(labels.labels, orders=ascending,
-                                      na_position=na_position)
+            indexer = labels._lexsort_indexer(orders=ascending,
```

Comment:
just do

```
indexer = lexsort_indexer(labels._get_labels_for_sorting(), orders=ascending,
                                      na_position=na_position)
```

so IOW rename ``pandas.indexes.multi._lexsort_indexer`` (your new routines) to ``_get_labels_for_sorting``